src/contrib/persistence/Akka.Persistence.Query.Sql/EventsByPersistenceIdPublisher.cs

```diff
@@ -109,6 +106,7 @@ protected bool Idle(object message)
         protected void Replay()
         {
             var limit = MaxBufferSize - Buffer.Length;
+            // TODO: use Ask<T> here to limit time similar to other queries?
```

Comment:
Right now the time to complete a recovery is not bounded by the caller for `_byPersistentId` queries - should we change that?

Reply:
So it turns out @Arkatufus that this is a major vulnerability in Akka.Persistence.Query.Sql currently - anything that goes wrong between here and the journal causes queries to die. Relying on the journal to do the right thing is too weak a guarantee. We have to use `Ask<T>` here and explicitly time out the request.

Reply:
Was able to reproduce this in the crush test just now

Reply:
Fixed this for two of the queries -have two more that need fixing too....